Petshop-iOS/Petshop-iOS/SceneDelegate.swift

```diff
@@ -16,10 +16,10 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         
         window = UIWindow(windowScene: windowScene)
         
-        let loginViewController = LoginViewController()
```

Comment:
Esse mesmo trecho de código também está em `Petshop-iOS/Petshop-iOS/AppDelegate.swift`, certo? Seria válido aplicar um "Don't repeat yourself"?

Reply:
O app delegate e o scene delegate são duas partes diferentes com duas funções diferentes, o appDelegate gerencia os eventos de ciclo de vida do app, enquanto o scene delegate gerencia as "scenes" (janelas) do app.

Reply:
São funções diferentes? Sim. Mas tu concorda que a principal parte do código útil faz a mesma coisa? Poderiam continuar a serem métodos diferentes (longe desse ser o caso citado), no entanto, se parte do comportamento esperado for igual, num seria legal extrair esse trecho que faz exatamente a mesma coisa? Algo do tipo "chamarLoginTela", só um exemplo.

 ```
    let loginVC = LoginViewController()
    let navigationController = UINavigationController(rootViewController: loginVC)

    window?.rootViewController = navigationController
    window?.makeKeyAndVisible()
```

Reply:
Eu dei uma pesquisada e vi que posso separar a funcoes, só tenho que manter os dois arquivos no projeto. removi do app delegate e deixei apenas no scene delegate.